### PR TITLE
Game state

### PR DESCRIPTION
**Pull request by ubeF**
Preparations for actions, events and duels
- Implemented a modifiable movement-queue (for skipping turn and repeating turns)
- Added helper functions for modification of all players at the same time
- Put all of this in a "State" singleton so the players and the helper functions can be accessed from anywhere in the code

### CODE REVIEW

File: main/main.gd [hunk 1]
```diff
@@ -77,6 +80,8 @@ func create_players(num):
 		player.update.connect(_on_update)
 		player.sprite.texture = sprites[i]
 		player.sprite.visibility_layer = 9
+		
+	State.queue = State.players.duplicate()
```
Comment (hfxbse) [inline main/main.gd]: Setting up the queue requires too much knowledge of the inner working of the state object for my liking. I prefer a small abstraction for this but not a significant concern tbh.

File: state.gd
Comment (hfxbse) [inline state.gd:1]: Is there actually a game scenario in which all an action is applied to all players?

File: state.gd [hunk 2]
```diff
@@ -0,0 +1,45 @@
+extends Node
+
+var players = []
+var queue = []
+var start
```
Comment (hfxbse) [inline state.gd:5]: The start field does not contain any state. I am not even sure it is used anywhere but I might missed it.

**Review by hfxbse — COMMENTED** [review]
Could the state be a resource instead of a node? The latter seems to be more semantically correct.

File: state.gd (hunk 2 above)
Comment (ubeF) [reply state.gd:5]: Will be needed as a teleport target for the "go to start action" not implemented yet

File: state.gd
Comment (ubeF) [reply state.gd:1]: Yes it's used for certain event cards

**Comment by ubeF** [comment]
> Could the state be a resource instead of a node? The latter seems to be more semantically correct.

I couldn't really find a clear answer on when to use resources vs auto-laded nodes, but: 
- i see no advantage of having to explicitly load the State in every scene i want to use it in
- the serialization features of resources are useless here
- having the State of the game accessible from everywhere once the game starts seems correct to me

So i'd leave it at the auto-loaded node

File: main/main.gd (hunk 1 above)
Comment (ubeF) [reply main/main.gd]: done

**Review by hfxbse — APPROVED** [review]

**Pull request merged**